### ISSUE TITLE
getresults() method of SuperpoweredOfflineAnalyzer class is not giving waveform information

### ISSUE THREAD

**Issue by shreyanshbangar** [superpoweredSDK/Low-Latency-Android-Audio-iOS-Audio-Engine#404]
I am using getresults() method of SuperpoweredOfflineAnalyzer class to get the detail about a audio file. But getresuls() method not giving complete detail.

I can get notes, waveformSize, overviewWaveform, overviewSize, averageDecibel, loudpartsAverageDecibel, peakDecibel, bpm, beatgridStartMs, keyIndex.

but averageWaveform, peakWaveform, lowWaveform, midWaveform, highWaveform are empty. i do not get any value in these.

I know "averageWaveform 150 points/sec waveform data displaying the average volume. Each sample is an unsigned char from 0 to 255." but where is the data???

I need these values to show the wave of audio.

Is there is any way other than this to show wave of complete audio file??

can you please tell how to get waveform details in getResult method. or can you tell any other way to show wave.

**Comment by superpoweredSDK** [comment]
What do you mean under "averageWaveform is empty"? Do you receive NULL for averageWaveform or... ?

**Comment by shreyanshbangar** [comment]
No, value of averageWaveform is not NULL. But when printing in log, averageWaveform, it's showing nothing.  "notes" is giving values but averageWaveform, peakWaveform, lowWaveform, midWaveform, highWaveform are not showing anything when printing in log.

**Comment by superpoweredSDK** [comment]
What kind of numbers are displayed when you log averageWaveform?

**Comment by shreyanshbangar** [comment]
It's not displaying anything. It's not NULL, but not showing anything.

**Comment by superpoweredSDK** [comment]
So you get the waveform data then. I don't understand what's the problem. It's clearly documented that each byte you get represents the amplitude in 150 bytes/second (normalized between 0 to 255).

**Comment by shreyanshbangar** [comment]
1)-This is what i m doing:-

   unsigned char *averageWaveform = NULL, *lowWaveform = NULL, *midWaveform = NULL, *highWaveform = NULL, *peakWaveform = NULL, *notes = NULL;

    int waveformSize, overviewSize, keyIndex;
    char *overviewWaveform = NULL;
    float loudpartsAverageDecibel, peakDecibel, bpm, averageDecibel, beatgridStartMs = 0;
    analyzer->getresults(&averageWaveform, &peakWaveform, &lowWaveform, &midWaveform, &highWaveform,
                         &notes, &waveformSize, &overviewWaveform, &overviewSize, &averageDecibel,
                         &loudpartsAverageDecibel, &peakDecibel, &bpm, &beatgridStartMs, &keyIndex);

    // Cleanup.
    delete decoder;
    delete analyzer;
    free(intBuffer);
    free(floatBuffer);


    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                        "wave averageWaveform is %s",
                        averageWaveform);
    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                        "wave notes is %s",
                        notes);
    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                        "wave peakWaveform is %s",
                        peakWaveform);
    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                        "wave lowWaveform is %s",
                        lowWaveform);
    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                        "wave midWaveform is %s",
                        midWaveform);
    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                        "wave highWaveform is %s",
                        highWaveform); 
2)-This is printed data in averagewaveform  :-
**wave averageWaveform is** 

3)-This is printed averageweveform when i do not use averageWaveform in getresults method, i.e. averageWaveform  is NULL.
**wave averageWaveform is (null)**

4)-This is the printed data in notes-:
**wave notes is �������������������������������������������������������������������������������������������������������������������������������������������������������������������H����**

I should get something like what i get in notes, but not getting anything. and averageWaveform is not null, because when its null it prints (null), but its not even printing (null).

**Comment by superpoweredSDK** [comment]
You are printing all as strings using the "%s", which is a mistake. You should print them as numbers, with an iteration.

**Comment by shreyanshbangar** [comment]
Thank you for your help. Got the values like this.

for(int i=0; i<waveformSize;i++)
    {
        
    __android_log_print((int) ANDROID_LOG_INFO, "SUPERPOWERED",
                            " averageWaveform %d ",
                            averageWaveform[i]);

    }
